[irwynliong] iP

irwynliong opened the pull request:
## Irwyn Chatbot 🤖
> " They slept on me 10 times!" 😶‍🌫️

Ever wanted a tasklist bot, that you could have used notion but instead **code it out yourself**? Introducing **Irwyn chatbot**!
It is :
- Not user friendly! 👍🏽
- ~~Effective~~ Not useful! 👍🏽
- Very Easy to use! 👎🏽

### All you need to do is 👇🏻

1. Clone the [Github repo](https://github.com/irwynliong/ip). 
2. Gradle run it. 🐘
3. List your task on the bot. 📄

And it just takes **_100000_** hours of your time to do this! ⌛ What a steal! 😎

### Upcoming updates:
- [x] Addition of GUI
- [x] Giving me headaches
- [x] Something

Heres a code snippet so that I meet the GFMD requirement, use `main` method:

```
public static void main(String[] args) {
        Irwyn.load();
        Irwyn.run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

irwynliong committed:
Level 0. Rename, Greet, Exit

irwynliong committed:
Level 1. Echo

irwynliong committed:
Level 2. Add, List

irwynliong committed:
Level 3. Mark as Done

irwynliong committed:
Level 4. ToDos, Events, Deadlines

irwynliong committed:
A-TextUiTesting

irwynliong committed:
Level 5. Handle Errors

irwynliong committed:
Level 6. Delete

irwynliong committed:
Level 6. Delete

irwynliong committed:
Revert "Level 6. Delete"

This reverts commit 4e3e288f3f0c0e7480eac329fee89aef63c5fba8.

irwynliong committed:
Revert "Revert "Level 6. Delete""

This reverts commit e72874297301170d70ed678883f298f05bc2ffbe.

irwynliong committed:
Divide classes into packages

irwynliong committed:
Merge pull request #1 from irwynliong/heads/A-Packages

Divide classes into packages

irwynliong committed:
Merge pull request #2 from irwynliong/add-gradle-support

Add gradle support

irwynliong committed:
Add gradle

irwynliong committed:
Add a saving feature

irwynliong committed:
Merge branch 'master' into branch-Level-7

irwynliong committed:
Merge pull request #3 from irwynliong/branch-Level-7

Add a saving feature

irwynliong committed:
Level-7

edit error

irwynliong committed:
Add a saving feature

irwynliong committed:
fixing

irwynliong committed:
Teach the chatbot how to understand dates and times

irwynliong committed:
Merge pull request #4 from irwynliong/branch-Level-8

Teach the chatbot how to understand dates and times

irwynliong committed:
packge the files

irwynliong committed:
OOP the classes

irwynliong committed:
Merge pull request #5 from irwynliong/branch-A-MoreOOP

OOP the classes

irwynliong committed:
Add JUnit tests

irwynliong committed:
Merge pull request #6 from irwynliong/branch-A-JUnit

 Add JUnit tests

irwynliong committed:
JAR file

irwynliong committed:
Java Doc files

irwynliong committed:
Merge pull request #7 from irwynliong/branch-A-JavaDoc

Java Doc files

irwynliong committed:
Java coding standards

irwynliong committed:
Merge pull request #8 from irwynliong/branch-A-CodingStandard

Java coding standards

irwynliong committed:
adding Find functionality.

irwynliong committed:
Merge pull request #9 from irwynliong/branch-Level-9

adding Find functionality.

irwynliong committed:
Merge branch 'master' of https://github.com/irwynliong/ip into branch-A-codingstandard

irwynliong committed:
fix my coding standards

irwynliong committed:
Merge pull request #11 from irwynliong/branch-fix-A-codingstandard

fix my coding standards

irwynliong committed:
fix my coding standards

irwynliong committed:
Merge pull request #12 from irwynliong/branch-A-codingstandard

fix my coding standards

chiageng commented on src/main/java/commands/Command.java:
```diff
@@ -0,0 +1,41 @@
+ * @version Week-3
+ */
+public abstract class Command {
+    boolean isExit;
```
Good naming convention for a boolean value.

chiageng commented on src/main/java/commands/CommandParser.java:
```diff
@@ -0,0 +1,46 @@
+                return new FindCommand(input);
+            default:
+                throw new TaskException();
+        }
```
I noticed a slight mistake here, in CS2103T coding guides, switch keyword and case keyword should have same indentation.
```
switch (command) {
case "bye":
....
....
}

chiageng commented on src/main/java/commands/ToDoCommand.java:
```diff
@@ -0,0 +1,42 @@
+package commands;
+
+import irwyn.tasks.TaskList;
+import irwyn.tasks.ToDo;
```
It is good and clear that you listed out imported class explicitly.

chiageng commented on src/main/java/irwyn/tasks/Task.java:
```diff
@@ -0,0 +1,94 @@
+
+    TaskType task;
+
+    public enum TaskType {
+        TODO,
+        EVENT,
+        DEADLINE
+    }
```
It is good that your constant variable name is all uppercase and is clear to other people.

chiageng reviewed (commented):
Overall it looks good. Keep it up.

LifHoshi reviewed (commented):
Well,  overall an excellent work of iP so far. Most of the obvious problems have been found out by another reviewer. By the way, I like the way that you create different folders to distinguish the use of purpose from other Java files.

irwynliong committed:
Added GUI for my chatbot.

irwynliong committed:
Merge pull request #13 from irwynliong/branch-Level-10

Added GUI for my chatbot.

irwynliong committed:
Add Checkstyle and use it to correct my code format.

irwynliong committed:
Merge pull request #14 from irwynliong/Branch-A-CheckStyle

Add Checkstyle and use it to correct my code format.

irwynliong committed:
add assert to Main and DialogBox.

irwynliong committed:
Merge pull request #15 from irwynliong/branch-A-Assertions

add assert to Main and DialogBox.

irwynliong committed:
Currently, there is no usage of assert.

It requires assert at Main and DialogBox so that users and the programme understands the assumptions that I made.

I used assert to ensure that all the assumption made are catch and the user using it will understand.

This is done so that my chatbot does not go into runtime error / null pointer error.

This is in addition to the A-Assertions commit made earlier

irwynliong committed:
Merge pull request #16 from irwynliong/branch-A-FullCommitMessage

Currently, there is no usage of assert.

irwynliong committed:
Follows the checkstyle code quality and followed the guideline for the code quality

irwynliong committed:
Merge pull request #17 from irwynliong/branch-A-CodeQuality

Adheres to the coding quality put out

irwynliong committed:
Add CI into the project.

irwynliong committed:
Currently there is no sorting feature

Addition of sorting feature to allow users to view their long list of tasks in a sorted manner

Add a new sort command in Commands directory
Add a sort method to Task manager
Make Task objects comparable by extending comparable interface
Add sort to the list of commands to the CommandParser

irwynliong committed:
Merge pull request #19 from irwynliong/branch-BCD-Extension

Currently there is no sorting feature

irwynliong committed:
fixing code quality

irwynliong committed:
fixing checkstyle

irwynliong committed:
fix coding quality

irwynliong committed:
Add User Guide

irwynliong committed:
Merge pull request #20 from irwynliong/branch-A-UserGuide

Add User Guide

irwynliong committed:
fix image location

irwynliong committed:
Update README.md

irwynliong committed:
fix image location

irwynliong committed:
Merge branch 'master' of github.com:irwynliong/ip

irwynliong committed:
Add Detect Duplicates function

This is to ensure no task is repeated in the taskList

irwynliong committed:
Merge pull request #21 from irwynliong/branch-C-DetectDuplicates

Add Detect Duplicates function

irwynliong committed:
fix task error.

irwynliong committed:
add title to bot

irwynliong committed:
fixing git standard

irwynliong committed:
fixing git standard

irwynliong committed:
fixing git standard

irwynliong committed:
fixing git standard

irwynliong committed:
fixing git standard

irwynliong committed:
fixing java version

irwynliong committed:
Fixing git standard 1

irwynliong committed:
Fixing git standard 2

irwynliong committed:
Fixing git standard 3

irwynliong committed:
Fixing git standard 4

irwynliong committed:
Fixing git standard 5
